seismostats/analysis/bvalue/more_positive.py

```diff
@@ -44,22 +44,20 @@ def calculate(self,
                         below this value are not considered). If None,
                         the cutoff is set to delta_m.
         '''
-
-        return super().calculate(magnitudes,
-                                 mc=mc,
-                                 delta_m=delta_m,
-                                 weights=weights,
-                                 dmc=dmc)
-
-    def _estimate(self, dmc: float) -> float:
-
-        self.dmc = dmc or self.delta_m
+        self.dmc: float = dmc if dmc is not None else delta_m
 
         if self.dmc < 0:
             raise ValueError('dmc must be larger or equal to 0')
-        elif self.dmc < self.delta_m and get_option('warnings') is True:
+
+        if self.dmc < delta_m and get_option('warnings') is True:
```

Comment:
how does this function know delta_m now?

Reply:
because its an argument of the function. It will only be set to the class instance (self) in the super().calculate(...), but I can already use it here. Remember this is an [overriden](https://www.geeksforgeeks.org/method-overriding-in-python/) class, so first this gets called, then I'm calling the super's method. It's a common and easy concept, but when you first see it, it can be a bit confusing.